### PR TITLE
feat: collect IPNI query results for public stats

### PR DESCRIPTION
**Pull request by bajtos**
This PR will be followed by another to expose this new data via HTTP (see https://github.com/filecoin-station/spark-stats).

Links:
- https://www.notion.so/pl-strflt/Moving-IPNI-queries-to-Spark-checkers-c58175c0f1c841e6bb82ac44a1472a98
- https://github.com/filecoin-station/spark/issues/40
- https://github.com/filecoin-station/roadmap/issues/84

### CODE REVIEW

File: migrations/004.do.indexer-stats.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+CREATE TABLE indexer_query_stats (
+  day DATE NOT NULL PRIMARY KEY,
+  total INT NOT NULL,
+  advertising_http INT NOT NULL
```
Comment (bajtos) [inline migrations/004.do.indexer-stats.sql:4]: Perhaps `http_advertised` would be a better column name?

File: migrations/004.do.indexer-stats.sql [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+);
```
Comment (bajtos) [inline migrations/004.do.indexer-stats.sql:5]: Do we want to keep track of the number of deals advertising Graphsync? On the one hand, it tells us how many SPs are advertising to IPNI (which is good) but not supporting HTTP yet (which may be easy to fix). OTOH, we don't really care about this metric, our ultimate goal is to have everybody advertising HTTP.

@patrickwoodhead thoughts?

File: migrations/004.do.indexer-stats.sql (hunk 1 above)
Comment (juliangruber) [reply migrations/004.do.indexer-stats.sql:4]: I think we can improve this by looking at the topology. We have query stats, and for each day we're tracking..the total number of stats advertising http? (I'm being naive here). We are tracking the total number of tasks and tasks with http advertisement, etc, but the notion of a task is missing.

What do you think about changing the name from `indexer_query_stats.advertising_http` to `task_indexer_stats.advertising_http` or `task_indexing_stats.advertising_http`?

**Review by juliangruber — CHANGES_REQUESTED** [review]

File: migrations/004.do.indexer-stats.sql (hunk 1 above)
Comment (bajtos) [reply migrations/004.do.indexer-stats.sql:4]: We are tracking daily stats of how many tasks (deals tested) have HTTP retrieval advertised in IPNI.

I can rename `_stats` to `_daily_stats`, but it's a long name and we track all stats with daily granularity (e.g. `retrieval_stats`).

I can rename the columns to `tasks_total` and `tasks_advertising_http`, but isn't that too long and a bit redundant?

Comment (bajtos) [reply migrations/004.do.indexer-stats.sql:4]: Oh, and we also want to move from "tasks" to "deals" in the near future (see https://github.com/filecoin-station/roadmap/issues/65).

Comment (bajtos) [reply migrations/004.do.indexer-stats.sql:4]: That leads me to the following column names:

- `deals_tested`
- `deals_advertising_http`

Comment (bajtos) [reply migrations/004.do.indexer-stats.sql]: I quite like these two names, I applied the rename in 2ac16e1.

PTAL again

Comment (bajtos) [reply migrations/004.do.indexer-stats.sql]: 89c1762 updates also variable names

File: lib/public-stats.js [hunk 3]
```diff
@@ -65,6 +68,35 @@ export const updateDailyParticipants = async (pgClient, participants) => {
...
 }
 
+const updateIndexerQueryStats = async (pgClient, honestMeasurements) => {
+  /** @type {Set<string>} */
+  const dealsWithHttpAdvertisement = new Set()
+  /** @type {Set<string>} */
+  const dealsWithIndexerResults = new Set()
+
+  for (const m of honestMeasurements) {
+    const dealId = getTaskId(m)
```
Comment (bajtos) [inline lib/public-stats.js:78]: The name `getTaskId` is not ideal, but also not that big of a deal IMO, so let's defer the cleanup/refactoring of this part until later (https://github.com/filecoin-station/roadmap/issues/65).

**Review by juliangruber — APPROVED** [review]

**Pull request merged**